### ISSUE TITLE
[ENHANCEMENT] Include Prometheus alerts in support bundles

### ISSUE THREAD

**Issue by bk201** [harvester/harvester#4993]
**Is your enhancement related to a problem? Please describe.**

Prometheus alerts can contain useful info while troubleshooting an issue.

<details>
<summary> Screenshot</summary>

![Prometheus-Time-Series-Collection-and-Processing-Server](https://github.com/harvester/harvester/assets/1691518/6265a5e0-3a28-47b7-ade9-c276c338f8e4)

</details>

**Describe the solution you'd like**

Include the Prometheus alerts in some kind of form if monitoring addon is enabled.

**Describe alternatives you've considered**

**Additional context**

**Comment by Yu-Jack** [comment]
@bk201 here is my thought.

When support bundle is created, we could put following data into support bundle tar and create same prometheus dashboard for troubleshooting.

- [Prometheus Snapshot ](https://prometheus.io/docs/prometheus/latest/querying/api/#snapshot), need to [enable web admin api](https://prometheus.io/docs/prometheus/latest/querying/api/#tsdb-admin-apis)
- [Prometheus Rules](https://prometheus.io/docs/prometheus/latest/querying/api/#rules)

That means you could check alert and graph in your local prometheus with snapshot and rules. There is an example I created few days ago. It will not collect new data since it's snapshot, but I think it's enough and convenient for troubleshooting with GUI.

![image](https://github.com/harvester/harvester/assets/6960289/fb54f1a2-ab83-478a-97fb-07315f2145af)
![image](https://github.com/harvester/harvester/assets/6960289/a365fbc5-0cd9-492f-a155-bcf4c877152f)

However, for alerting, we need to get current alert status and save it. Because if we use snapshot, the alert will disappear in the end due to no new coming data.

**Comment by innobead** [comment]
> However, for alerting, we need to get current alert status and save it. Because if we use snapshot, the alert will disappear in the end due to no new coming data.

Is there any challenge or issue in collecting and storing alert statuses?

**Comment by Yu-Jack** [comment]
> > However, for alerting, we need to get current alert status and save it. Because if we use snapshot, the alert will disappear in the end due to no new coming data.
> 
> Is there any challenge or issue in collecting and storing alert statuses?

I find out we could directly use `ALERT` query to get all alerts like below. Then, you could specify time range and severity to query old alerts for troubleshooting.

<details>
<summary>click me to see screenshot</summary>

![image](https://github.com/harvester/harvester/assets/6960289/99106b6b-0747-4edc-9821-0f56d5bf0033)

</details>

**Comment by Yu-Jack** [comment]
In the first version of this feature, we could try to just get the **current alert** to see it's useful or not after discussion with @bk201.  But, I might try to fetch more alters, for example, I could get 30 minutes alerts. Depend what we need.

However, I still have a quick PoC could download support bundle with prometheus snapshots in [this branch](https://github.com/Yu-Jack/support-bundle-kit/blob/8ba70230002a8ef3409a42ff12ff1b1d56a1017a/pkg/manager/manager.go#L481-L520). But, the prometheus snapshots is quite big. Anyway, I left this information here for one day might need this.

**Comment by TachunLin** [comment]
Verified **fixed** on `master-8b80f82-head` (24/07/19). Close this issue. 

### Result
<details><summary>$\color{green}{\textsf{PASS}}$ Can find new Prometheus alerts json file in support bundle after enabling rancher-monitoring addon $~~$</summary>

  1. Can generate support bundle correctly
  2. New `prometheus-alerts.json` file exists in the generated support bundle
  ```
  ~/Documents/Harvester/tools/support_bundle/supportbundle_eb6de629-a9c3-46c6-9922-cdd1ffc27387_2024-07-22T07-14-46Z> ls
  bundleGenerationError.log  logs  metadata.yaml  nodes  prometheus-alerts.json  yamls
  ```

  3. Can open the `prometheus-alerts.json` and contain required information
  
  ```
	  {
		  "activeAt": "2024-07-22T05:56:41.152925493Z",
		  "Annotations": {
			  "description": "32.99% throttling of CPU in namespace cattle-monitoring-system for container prometheus in pod prometheus-rancher-monitoring-prometheus-0.",
			  "runbook_url": "https://runbooks.prometheus-operator.dev/runbooks/kubernetes/cputhrottlinghigh",
			  "summary": "Processes experience elevated CPU throttling."
		  },
		  "Labels": {
			  "alertname": "CPUThrottlingHigh",
			  "container": "prometheus",
			  "namespace": "cattle-monitoring-system",
			  "pod": "prometheus-rancher-monitoring-prometheus-0",
			  "severity": "info"
		  },
		  "State": "firing",
		  "Value": "3.298722044728435e-01"
	  },
  ```


 </details>


<details><summary>$\color{green}{\textsf{PASS}}$ Can't find new Prometheus alerts json file in support bundle after disabled the rancher-monitoring addon $~~$</summary>

  1. Can generate support bundle correctly
  2. New `prometheus-alerts.json` file not exists in the generated support bundle

    ```
    davidtclin@localhost:~/Documents/Harvester/Issues/4993/disable-monitoring/supportbundle_eb6de629-a9c3-46c6-9922-cdd1ffc27387_2024-07-22T08-51-00Z> ls
    bundleGenerationError.log  logs  metadata.yaml  nodes  yamls
    ```

 </details>


### Test Information
* Test Environment: Four nodes harvester on remote baremetal machines
* Harvester version: `master-8b80f82-head`

### Verify Steps
<details><summary>Check the Prometheus alert file in support bundle </summary>

  1. Enable the rancher-monitoring addon
  2. Check the monitoring dashboard and vm metrics display well 
  3. Open the settings
  4. Change the support-bundle-image version to `v0.0.40`
  
    ![image](https://github.com/user-attachments/assets/e1b999cd-f033-4ecc-b471-5ad552a899f8)
  5. Generate the support bundle 
  6. Unzip the support bundle zip file 
  7. Check the `prometheus-alerts.json` file exists
  8. Open the file and check the content

</details>
